pkg/synopsysctl/cmd_migrate.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright (C) 2019 Synopsys, Inc.
+
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements. See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership. The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License. You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied. See the License for the
+specific language governing permissions and limitations
+under the License.
+*/
+
+package synopsysctl
+
+import (
+	"fmt"
+
+	"github.com/blackducksoftware/synopsys-operator/pkg/util"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// migrateCmd migrates a resource before a synopsys operator upgrade
+var migrateCmd = &cobra.Command{
+	Use:   "migrate",
+	Short: "Migrate a Synopsys resource before upgrading the operator",
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return fmt.Errorf("not a valid command")
+	},
+}
+
+// migrateBlackduckCmd migrates a blackduck
+var migrateBlackduckCmd = &cobra.Command{
+	Use:     "blackduck NAMESPACE...",
+	Example: "synopsysctl migrate blackduck [namespace]\nsynopsysctl migrate blackduck [namespace1] [namespace2] [namespace3]",
```

Comment:
These should be blackduck names, not namespaces right?  From a user perspective that give a name for the instance in the other commands iirc.

Reply:
Well, currently all of our commands, `create`, `update`, `delete` use the convention: `synopsysctl COMMAND blackduck NAMESPACE [flags]`.  We can change that, but that's a different issue than this PR.

Reply:
We'll need to change them all from NS to Name because of Namespaced mode.  That'll need to be a separate PR.

Reply:
Created #431